### PR TITLE
Added instructions in readme

### PR DESCRIPTION
**Pull request by skleanthous**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,69 @@
...
+
+### Commands accepted by SLMM client
+1. TL -> Turn Left
+2. TR -> Turn Right
+3. MF -> Move Forward
+4. ML -> Mow Lawn
+
+### Action methods exposed by SLMM
+1. /lawn Method: POST. Payload: {"StartX": int, "StartY": int, "SizeX": int, "SizeY": int}
+2. /location Method: PUT. Payload: {"Units": int}
```
Comment (sagarwdfc) [inline README.md]: Don't we need to specify which direction to turn , right or left?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,69 @@
...
+3. /location Method: GET. Reply: {"X": int, "Y": int} -> returns the current location of the SLMM
+4. /mower Method: PUT Payload {"On": boolean}
+
+### Deliverables
+First provide the deliverables for Step 1 independantly. Then provide one console application that represents the application that is run in the SLMM and one console application that accepts command input from console and communicates it to the SLMM application via Http along with all supporting libraries. Again, you should provide access to an online repository that hosts all code for all applications, libraries and test projects (we recommend this to be in another branch than the one for step 1). The readme should have all necessary information on how to build, run and test the solution.
+
+### Acceptance criteria
+1. The SLMM never goes outside of the dimensions of the garden as supplied during startup
+2. In the SLMM application, it is expected that action methods will log when they are called, start of operations and end of operations should be logged as well.
```
Comment (sagarwdfc) [inline README.md]: "action methods will be logged when they are called"

File: README.md [hunk 3]
```diff
@@ -1,2 +1,69 @@
...
+2. TR -> Turn Right
+3. MF -> Move Forward
+4. ML -> Mow Lawn
+
+### Deliverable
+One console application that represents the application as described. You should provide access to an online repository, based on this one that contains all source code for this application, along with complete instructions on how to build your application locally. If any tests or libraries were written, please provide code for all and full instructions on how to run the build and tests.
+
+### Acceptance criteria
+1. The SLMM never goes outside of the dimensions of the garden as supplied during startup
+2. The output of the SLMM after it finished each action must be in the format: “{Time} – {Action taken place} – {current location of SLMM}”
```
Comment (sagarwdfc) [inline README.md]: The output for this step would not show the end time of an action as it as been specified in the step2 acceptance criteria?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,69 @@
...
+2. Since now this is an ASP.Net Web API, the size of the lawn should be passed in through an action method specifically for this purpose.
+3. All commands should be done asynchronously and all action methods should just return a correlation id.
+4. The commands should be done on a FIFO manner.
+5. There should be two applications, one that runs in SLMM and one that represents the client and will be used by administrator(s) to control the SLMM. See below for accepted commands and exposed action methods.
+
+### Commands accepted by SLMM client
+1. TL -> Turn Left
+2. TR -> Turn Right
+3. MF -> Move Forward
+4. ML -> Mow Lawn
```
Comment (sagarwdfc) [inline README.md]: Does the "/mower" method sends  the Mow lawn command or is this just to start the lawn mower. If so how do we send the Mow lawn command from the http client.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,69 @@
...
+
+### Action methods exposed by SLMM
+1. /lawn Method: POST. Payload: {"StartX": int, "StartY": int, "SizeX": int, "SizeY": int}
+2. /location Method: PUT. Payload: {"Units": int}
+3. /location Method: GET. Reply: {"X": int, "Y": int} -> returns the current location of the SLMM
+4. /mower Method: PUT Payload {"On": boolean}
```
Comment (sagarwdfc) [inline README.md]: Can this be named /mowLawn , it looks like we just turning on the lawn mower and not doing anything else.

**Review by sagarwdfc — COMMENTED** [review]
Just some minor comments and few questions

File: README.md [hunk 6]
```diff
@@ -1,2 +1,69 @@
...
+
+Step 1 – all server-side positions
+----------------------------------
+Assume we are interested in creating a smart lawn mowing machine called SLMM (smart lawn mowing machine). Your task is to create the software that will run in the SLMM itself and will be responsible of doing the following actions:
+
+1. Turn the SLMM 90o left or right -> This should take 15 seconds to do
+2. Move forward by one position -> This takes 30 seconds to do
+3. Mow the lawn at the current position -> This takes 120 seconds to do
+
+The application you will create should be able to accept the above commands and do them. Since you do not have access to the actual hardware, for now the only thing required is to accept input from console and output the result (the action being done along with the position of the SLMM) in the console. During application startup, the SLMM is given a set of dimensions (passed in as command line parameters) that represent the width and length of the garden that the SLMM is in.
```
Comment (skleanthous) [inline README.md]: I'm not really OK with your version either tbh. I think a better version would be to say "and write the status at the end of the action in the console (write the completed action along with the position of the SLMM)"

File: README.md (hunk 3 above)
Comment (skleanthous) [inline README.md]: The output should only happen at the end of the action in step 1.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,69 @@
...
+3. MF -> Move Forward
+4. ML -> Mow Lawn
+
+### Deliverable
+One console application that represents the application as described. You should provide access to an online repository, based on this one that contains all source code for this application, along with complete instructions on how to build your application locally. If any tests or libraries were written, please provide code for all and full instructions on how to run the build and tests.
+
+### Acceptance criteria
+1. The SLMM never goes outside of the dimensions of the garden as supplied during startup
+2. The output of the SLMM after it finished each action must be in the format: “{Time} – {Action taken place} – {current location of SLMM}”
+3. The commands to be accepted in this version are:
```
Comment (skleanthous) [inline README.md]: They are above, I will remove this line

File: README.md (hunk 4 above)
Comment (skleanthous) [inline README.md]: This is to mow the lawn at the particular location and we are expected to send this command over every time. I will update the documentation about this.

File: README.md (hunk 1 above)
Comment (skleanthous) [inline README.md]: This is to move. I will update documentation to make this clearer.

File: README.md (hunk 5 above)
Comment (skleanthous) [inline README.md]: I think mower is better as it is turning the mower on to mow the particular location only. I will update docs.

File: README.md (hunk 2 above)
Comment (skleanthous) [inline README.md]: I am trying to write in an active, current verse. So I think maybe "should log when they get called"?

File: README.md [hunk 8]
```diff
@@ -1,2 +1,80 @@
...
+
+Step 1 – all server-side positions
+----------------------------------
+Assume we are interested in creating a smart lawn mowing machine called SLMM (smart lawn mowing machine). Your task is to create the software that will run in the SLMM itself and will be responsible for doing the following actions:
+
+1. Turn the SLMM 90o left or right -> This should take 15 seconds to do
+2. Move forward by one position -> This takes 30 seconds to do
+3. Mow the lawn at the current position -> This takes 120 seconds to do
+
+Please just sleep the thread for the specified duration above to represent\simulate work being done.
```
Comment (sagarwdfc) [inline README.md]: "Please make the thread go to sleep" sounds a bit better

**Review by sagarwdfc — COMMENTED** [review]
minor comment

**Comment by sagarwdfc** [comment]
Review done. Please merge

**Pull request merged**